### DISCUSSION TITLE
[Discussion] How to estimate the time complexity of the webpack build/compilation?

### DISCUSSION THREAD

**Discussion by suguanYang** [webpack/webpack#16116] [Ideas]
Let's say we use typescript on a project, it has 1000 lines of code, and there have 10 esmodule dependencies, and we assume the base compilation complexity of the typescript loader is `n`, and the per dependency bundle is `m`, can we estimate a approximate time complexity of webpack build this project to a single js file? Is there a formula like `<line of code> * <loader complexity> + <number of dependecies> * <bundle complexity>` so the final time complexity of this project would be `1000 * n + 10 * m`

**Comment by vankop** [comment]
very unlikely.. it depends on your webpack config a lot.. + `<line of code>` is wrong, it depends on your code..
